kubectl-plugin/pkg/cmd/create/create_cluster.go

```diff
@@ -0,0 +1,160 @@
+	cmd.Flags().StringVar(&options.image, "image", options.image, "Ray image to use in the Ray Cluster yaml")
+	cmd.Flags().StringVar(&options.headCPU, "head-cpu", "500m", "Number of CPU for the ray head. Default to 500m")
+	cmd.Flags().StringVar(&options.headMemory, "head-memory", "1Gi", "Amount of memory to use for the ray head. Default to 1Gi")
+	cmd.Flags().StringVar(&options.workerGrpName, "worker-grp-name", "default-group", "Name of the worker group for the Ray Cluster")
+	cmd.Flags().Int32Var(&options.workerReplicas, "worker-replicas", 1, "Number of the worker group replicas. Default of 1")
+	cmd.Flags().StringVar(&options.workerCPU, "worker-cpu", "500m", "Number of CPU for the ray worker. Default to 500m")
+	cmd.Flags().StringVar(&options.workerMemory, "worker-memory", "1Gi", "Amount of memory to use for the ray worker. Default to 1Gi")
+	cmd.Flags().BoolVar(&options.dryRun, "dry-run", false, "Will not apply the generated cluster and will print out the generated yaml")
+
+	options.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (options *CreateClusterOptions) Complete(cmd *cobra.Command, args []string) error {
+	if *options.configFlags.Namespace == "" {
+		*options.configFlags.Namespace = "default"
+	}
+
+	if len(args) != 1 {
+		return cmdutil.UsageErrorf(cmd, "%s", cmd.Use)
+	}
+	options.clusterName = args[0]
+
+	if options.image == "" {
+		options.image = fmt.Sprintf("rayproject/ray:%s", options.rayVersion)
+	}
+
+	return nil
+}
+
+func (options *CreateClusterOptions) Validate() error {
+	config, err := options.configFlags.ToRawKubeConfigLoader().RawConfig()
+	if err != nil {
+		return fmt.Errorf("Error retrieving raw config: %w", err)
+	}
+	if len(config.CurrentContext) == 0 {
+		return fmt.Errorf("no context is currently set, use %q to select a new one", "kubectl config use-context <context>")
+	}
+
+	return nil
+}
+
+func (options *CreateClusterOptions) Run(ctx context.Context, factory cmdutil.Factory) error {
+	k8sClient, err := client.NewClient(factory)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %w", err)
+	}
+
+	// Will generate yaml file
+	fmt.Printf("Generating Ray Cluster yaml...\n")
+	rayClusterObject := generation.RayClusterYamlObject{
+		Namespace:      *options.configFlags.Namespace,
+		ClusterName:    options.clusterName,
+		RayVersion:     options.rayVersion,
+		Image:          options.image,
+		HeadCPU:        options.headCPU,
+		HeadMemory:     options.headMemory,
+		WorkerGrpName:  options.workerGrpName,
+		WorkerReplicas: options.workerReplicas,
+		WorkerCPU:      options.workerCPU,
+		WorkerMemory:   options.workerMemory,
+	}
+
+	rayClusterac, err := rayClusterObject.GenerateRayClusterApplyConfig()
+	if err != nil {
+		return err
+	}
+
+	// If dry run is enabled, it will call the yaml converter and print out the yaml
+	if options.dryRun {
+		fmt.Printf("Generated Yaml: \n")
```

Comment:
Consider removing this line, it may not be friendly for tools that pipe the output of this command into another CLI

Reply:
There were a few other lines I was printing out as well. Removed.